src/workerd/api/pyodide/pyodide.c++

```diff
@@ -7,6 +7,17 @@
 
 namespace workerd::api::pyodide {
 
+kj::Maybe<kj::Array<unsigned char>> pyodideBundleDataGlobal = kj::none;
+kj::Maybe<kj::Own<capnp::FlatArrayMessageReader>> pyodideBundleReaderGlobal = kj::none;
+kj::Maybe<jsg::Bundle::Reader> pyodideBundleGlobal = kj::none;
+
+void setPyodideBundleData(kj::Array<unsigned char> data) {
+  pyodideBundleReaderGlobal = kj::heap<capnp::FlatArrayMessageReader>(kj::arrayPtr(
+        reinterpret_cast<const capnp::word*>(data.begin()), data.size() / sizeof(capnp::word)));
+  pyodideBundleGlobal = KJ_REQUIRE_NONNULL(pyodideBundleReaderGlobal)->getRoot<jsg::Bundle>();
+  pyodideBundleDataGlobal = kj::mv(data);
+}
```

Comment:
I don't feel too strongly about this, but I think it might be worth to reduce the number of globals here. Creating a reader should be fast, so maybe you could instead only keep the `pyodideBundleDataGlobal` global and create the `jsg::Bundle::Reader` on-demand?

Reply:
I don't understand the ownership semantics here, I'm concerned that the `jsg::Bundle::Reader` doesn't own its data and instead the data is owned by one of `pyodideBundleReaderGlobal` or `pyodideBundleDataGlobal`, so if I let either of those go out of scope while the `ModuleRegistry` object is still around, then when a file is imported, we'll read from freed data.

Reply:
Looking at other examples of `capnp::FlatArrayMessageReader` it seems it is important to keep it all alive. But it looks like you can reduce the number of globals at least a little bit by using `attach`: https://github.com/cloudflare/workerd/blob/main/src/workerd/server/workerd.c%2B%2B#L991